airflow/api_fastapi/core_api/routes/public/dag_stats.py

```diff
@@ -0,0 +1,79 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from fastapi import Depends
+from sqlalchemy import func, select
+from sqlalchemy.orm import Query, Session
+from typing_extensions import Annotated
+
+from airflow.api_fastapi.common.db.common import (
+    get_session,
+)
+from airflow.api_fastapi.common.router import AirflowRouter
+from airflow.api_fastapi.core_api.openapi.exceptions import create_openapi_http_exception_doc
+from airflow.api_fastapi.core_api.serializers.dag_stats import (
+    DagStatsCollectionResponse,
+    DagStatsResponse,
+    DagStatsStateResponse,
+)
+from airflow.models.dagrun import DagRun
+from airflow.utils.state import DagRunState
+
+dag_stats_router = AirflowRouter(tags=["DagStats"], prefix="/dagStats")
+
+
+@dag_stats_router.get(
+    "/",
+    responses=create_openapi_http_exception_doc([400, 401, 403, 404, 422]),
+)
+async def get_dag_stats(
+    session: Annotated[Session, Depends(get_session)],
+    dag_ids: str | None = None,
```

Comment:
We can do that, but that'll change the request format a bit as follows:

- Current (old API) request: `/dagStats?dag_ids=dag1,dag2`
- If we use `list[str] | None = Query(None)`, then request will be: `/dagStats?dag_ids=dag1&dag_ids=2` instead of `/dagStats?dag_ids=dag1,dag2`.

Let me know if this change in request format is fine, I'll update it to `dag_id` accordingly and create a common parameter. Thank you! 😃

Reply:
Yes, FastAPI handles list natively in the exploded way. This will be one of the API breaking change I believe and I don't think we want to manually handle list to make that backward compatible.

HAving `?param=item1&param=item2` is the way to go with FastAPI.

Reply:
Done.

Reply:
>You can make a common parameter paramters because this filtering based on dag_ids list is pretty common in the API.

This done too. Also have moved the query to the db commons module. Updated this PR.